Modules/posixmodule.c

```diff
@@ -376,6 +378,13 @@ corresponding Unix manual entries for more information on calls.");
 #  define fsync _commit
 #endif  /* ! __WATCOMC__ || __QNX__ */
 
+// iOS/tvOS/watchOS *define* some POSIX methods,
+// but raise a compiler error if they are used.
+#if TARGET_OS_IPHONE
+#  undef HAVE_GETGROUPS
```

Comment:
If it's a compiler error, as opposed to a runtime error, then autoconf should be able to detect the problem. The comment should explain why that wasn't sufficient.